test/features/auth/bloc/signup/signup_bloc_test.dart

```diff
@@ -0,0 +1,182 @@
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (invalid)] when username field changes to invalid input",
+        build: () => SignUpFormBloc(mocksAuthRepository),
+        act: (SignUpFormBloc bloc) {
+          bloc.add(UsernameChanged(username: "32342"));
+        },
+        expect: () {
+          final username = Username.dirty("32342");
+          return [
+            SignUpFormState(
+              username: Username.pure("32342"),
+              status: Formz.validate([username]),
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (valid)] when email field changes to valid input",
+        build: () => SignUpFormBloc(mocksAuthRepository),
+        act: (SignUpFormBloc bloc) {
+          bloc.add(EmailChanged(email: "email@gmail.com"));
+        },
+        expect: () {
+          final email = Email.dirty("email@gmail.com");
+          return [
+            SignUpFormState(
+              email: email,
+              status: Formz.validate([email]),
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (invalid)] when email field changes to invalid input",
+        build: () => SignUpFormBloc(mocksAuthRepository),
+        act: (SignUpFormBloc bloc) {
+          bloc.add(EmailChanged(email: "email"));
+        },
+        expect: () {
+          final email = Email.dirty("email");
+          return [
+            SignUpFormState(
+              email: Email.pure("email"),
+              status: Formz.validate([email]),
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (valid)] when password field changes to valid input",
+        build: () => SignUpFormBloc(mocksAuthRepository),
+        act: (SignUpFormBloc bloc) {
+          bloc.add(PasswordChanged(password: "1fygfgy5"));
+        },
+        expect: () {
+          final password = Password.dirty("1fygfgy5");
+          return [
+            SignUpFormState(
+              password: password,
+              status: Formz.validate([password]),
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (invalid)] when password field changes to invalid input",
+        build: () => SignUpFormBloc(mocksAuthRepository),
+        act: (SignUpFormBloc bloc) {
+          bloc.add(PasswordChanged(password: "1fygf"));
+        },
+        expect: () {
+          return [
+            SignUpFormState(
+              password: Password.pure("1fygf"),
+              status: FormzStatus.invalid,
+            )
+          ];
+        });
+
+    blocTest(
+        "emits [SignUpFormState (valid), SignUpFormState (In Progress), SignUpFormState (success)] when form is submitted with valid inputs",
+        build: () {
+      when(mocksAuthRepository.getRemoteProvider)
+          .thenAnswer((_) => mockAuthRemoteDataProvider);
+      when(mockAuthRemoteDataProvider.createUser(
```

Comment:
Change to authRepository.createRemoteUser

Reply:
@seraph14 Great review